python/fusion_engine_client/utils/construct_utils.py

```diff
@@ -6,6 +7,31 @@
 from .enum_utils import IntEnum
 
 
+class FixedPointAdapter(Adapter):
+    def __init__(self, scale, *args, invalid=None):
+        super().__init__(*args)
+        self.scale = scale
+        self.invalid = invalid
+
+        is_signed = self.subcon.fmtstr.islower()
+        if self.invalid is not None and is_signed:
+            max_negative_mag = 2 ** (self.subcon.sizeof() * 8 - 1)
+            if self.invalid == max_negative_mag:
+                self.invalid = -self.invalid
+
+    def _decode(self, obj, context, path):
+        if obj == self.invalid:
+            return math.nan
+        else:
+            return obj * self.scale
```

Comment:
Small thing, but I guess this might have some slightly weird behavior since this will sometimes return an int, and sometimes a float depending on the scale. Probably worth casting to a float for consistency.

Reply:
Makes sense